File: samples/sample-pdk-pipeline-ts/src/pipeline-stack.ts
import { Stack, StackProps } from 'aws-cdk-lib';
import { pdk_pipeline } from 'aws-prototyping-sdk';
import { Construct } from 'constructs';

export class PipelineStack extends Stack {
  readonly pipeline: pdk_pipeline.PDKPipeline;

  constructor(scope: Construct, id: string, props: StackProps) {
    super(scope, id, props);

    this.pipeline = new pdk_pipeline.PDKPipeline(this, 'ApplicationPipeline', {
      primarySynthDirectory: 'packages/infra/cdk.out',
      repositoryName: 'monorepo',
      publishAssetsInParallel: false,
      crossAccountKeys: true,
      synth: {},
    });
  }
}

File: samples/sample-pdk-pipeline-java/src/main/java/software/aws/PipelineStack.java
package software.aws;

import org.jetbrains.annotations.Nullable;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.pipelines.FileSet;
import software.amazon.awscdk.pipelines.IFileSetProducer;
import software.aws.awsprototypingsdk.pdk_pipeline.PDKPipeline;
import software.aws.awsprototypingsdk.pdk_pipeline.PDKPipelineProps;
import software.constructs.Construct;

public class PipelineStack extends Stack {
    private PDKPipeline pipeline;

    public PipelineStack(Construct scope, String id, @Nullable StackProps props) {
        super(scope, id, props);

        this.pipeline = new PDKPipeline(this, "ApplicationPipeline", PDKPipelineProps.builder()
                .primarySynthDirectory("packages/infra/cdk.out")
                .repositoryName("monorepo")
                .publishAssetsInParallel(false)
                .crossAccountKeys(true)
                .synth(new IFileSetProducer() {
                    @Override
                    public @Nullable FileSet getPrimaryOutput() {
                        return null;
                    }
                })
                .build());
    }

    public PDKPipeline getPipeline() {
        return this.pipeline;
    }
}
